yargs.js

```diff
@@ -301,42 +301,58 @@ function Yargs (processArgs, cwd, parentRequire) {
     return self
   }
 
-  self.demand = self.required = self.require = function (keys, max, msg) {
+  self.demandOption = self.demand = self.required = self.require = function (keys, max, msg) {
```

Comment:
@lrlna Yeah that sounds right to me. I'm just meaning, if you took out logic of the `demandOption` compared to the old `demand`, `demandOption` doesn't have all the features the old `demand` did.

If folks now (or rather then) download 7.x or something and get no error because `.demand()` still exists, but something like `.demand(1)` doesn't work anymore, that might cause confusion.

Maybe I'm just not getting something, but apart from that, it might also be bad to keep it because of the naming stuff. Like, `demandOption` and `demandCommand` are very expressive names, but I would find it hard to remember which `demand` was an alias of. What do you think?